src/metabase/email/messages.clj

```diff
@@ -58,39 +58,45 @@
 ;; HACK: temporary workaround to postal requiring a file as the attachment
 (defn- write-byte-array-to-temp-file
   [img-bytes]
-  (let [file (java.io.File/createTempFile "metabase_pulse_image_" ".png")
-        fos (new java.io.FileOutputStream file)]
-    (.deleteOnExit file)
-    (.write fos img-bytes)
-    (.close fos)
+  (let [file (doto (java.io.File/createTempFile "metabase_pulse_image_" ".png")
+               .deleteOnExit)]
+    (with-open [fos (java.io.FileOutputStream. file)]
+      (.write fos (bytes img-bytes)))
     file))
 
 (defn- find-byte-array
   [needle haystack]
-  (first (keep-indexed #(when (java.util.Arrays/equals %2 needle) %1) haystack)))
+  (first (keep (fn [item]
+                 (when (java.util.Arrays/equals (bytes item) (bytes needle))
+                   true))
+               haystack)))
```

Comment:
You can just use `keep` instead of `keep-indexed` if you're not actually using the index for anything